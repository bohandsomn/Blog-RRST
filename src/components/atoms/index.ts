export * from './Img'
export * from './P'
export * from './Icon'
export * from './Input'
export * from './Textarea'
export * from './Button'
export * from './Div'
export * from './Ul'
export * from './Li'
export * from './Label'
export * from './Form'