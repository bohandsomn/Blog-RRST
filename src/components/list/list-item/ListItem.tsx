import React, { useContext } from 'react'
import Toggle from '../../../context/toggle'
import useAppStyles from '../../../hooks/useAppStyles'
import { Div, Li, MainParagraph, SecondaryParagraph } from '../../atoms'

export const ListItem: React.FC<Props> = ({ children, right, left }) => {
    const {value, handleSetTrue, handleSetFalse} = useContext(Toggle)
    const className = useAppStyles(
        'border-radius--border-radius', 
        'color--main-color', 
        'display-flex', 
        'align-items-center', 
        'justify-content-space-between', 
        'padding-top-10', 
        'padding-right-10', 
        'padding-bottom-10', 
        'padding-left-10', 
        'cursor-pointer', 
        value ? 'background-color--main-bg-color' : undefined
    )
    const rightClassName = useAppStyles('margin-left-10')
    const leftClassName = useAppStyles('margin-right-10')

    return (
        <Li className={className} onMouseEnter={handleSetTrue} onMouseLeave={handleSetFalse}>
            {left && <Div className={leftClassName}>{left}</Div>}
            {value 
                ? <SecondaryParagraph>{children}</SecondaryParagraph>
                : <MainParagraph>{children}</MainParagraph>
            }
            {right && <Div className={rightClassName}>{right}</Div>}
        </Li>
    )
}

type Props = NonNullable<typeof Li.defaultProps> & {
    right?: JSX.Element
    left?: JSX.Element
}
