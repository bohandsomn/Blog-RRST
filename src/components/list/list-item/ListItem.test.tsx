import { fireEvent, render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import { ListItem } from './ListItem'
import { Account } from '../../atoms'
import ToggleProvider from '../../../provider/toggle'

describe('ListItem', () => {
    test('should renders the element', () => {
        render(<ToggleProvider><ListItem>ListItem</ListItem></ToggleProvider>)
        const element = screen.getByText(/ListItem/)
        expect(element).toBeInTheDocument()
    })

    test('should containg svg', () => {
        render(
            <ToggleProvider>
                <ListItem right={<Account data-testid="svg" />}>
                    ListItem
                </ListItem>
            </ToggleProvider>
        )
        const element = screen.getByTestId('svg')
        expect(element).toBeInTheDocument()
    })
    
    test('should changes classList', () => {
        render(
            <ToggleProvider>
                <ListItem right={<Account />}>
                    ListItem
                </ListItem>
            </ToggleProvider>
        )
        const initialElement = screen.getByText(/ListItem/)
        fireEvent.mouseEnter(initialElement)
        expect(initialElement).toMatchSnapshot()

        const afterEnterElement = screen.getByText(/ListItem/)
        fireEvent.mouseLeave(afterEnterElement)
        expect(afterEnterElement).toMatchSnapshot()

        const afterLeaveElement = screen.getByText(/ListItem/)
        expect(afterLeaveElement).toMatchSnapshot()
    })
})