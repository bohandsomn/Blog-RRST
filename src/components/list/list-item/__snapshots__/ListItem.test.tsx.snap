// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem should changes classList 1`] = `
<p
  class="sc-gswNZR gyGtCJ"
>
  ListItem
</p>
`;

exports[`ListItem should changes classList 2`] = `
<p
  class="sc-gswNZR cSeRYA"
>
  ListItem
</p>
`;

exports[`ListItem should changes classList 3`] = `
<p
  class="sc-gswNZR gyGtCJ"
>
  ListItem
</p>
`;
